### PR TITLE
[SPARK-34699][SQL] 'CREATE OR REPLACE TEMP VIEW USING' should uncache correctly

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

This PR proposes:
  1. `CREATE OR REPLACE TEMP VIEW USING` should use `TemporaryViewRelation` to store temp views.
  2. By doing #1, it fixes the issue where the temp view being replaced is not uncached.

### Why are the changes needed?

This is a part of an ongoing work to wrap all the temporary views with `TemporaryViewRelation`: [SPARK-34698](https://issues.apache.org/jira/browse/SPARK-34698).

This also fixes a bug where the temp view being replaced is not uncached.

### Does this PR introduce _any_ user-facing change?

Yes, the temp view being replaced with `CREATE OR REPLACE TEMP VIEW USING` is correctly uncached if the temp view is cached.

### How was this patch tested?

Added new tests.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala [hunk 1]
```diff
@@ -1509,4 +1509,48 @@ class CachedTableSuite extends QueryTest with SQLTestUtils
...
+      assert(spark.catalog.isCached(viewName))
+
+      // Replacing with the same relation. The cache shouldn't be uncached.
+      sql(s"CREATE OR REPLACE $tempViewStr ${ident.table} USING parquet OPTIONS (path '$path1')")
+      assert(spark.catalog.isCached(viewName))
+
+      // Replacing with a different relation. The cache should be cleared.
+      sql(s"CREATE OR REPLACE $tempViewStr ${ident.table} USING parquet OPTIONS (path '$path2')")
+      assert(!spark.catalog.isCached(viewName))
+      assert(spark.sharedState.cacheManager.isEmpty)
```
Comment (imback82) [inline sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala:1553]: The existing code fails here because the temp view is not uncached.

**Comment by HyukjinKwon** [comment]
cc @MaxGekk @sunchao @cloud-fan FYI

**Comment by cloud-fan** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala:1553]: Can we remove this line? This looks a bit fragile as `cacheManager` is global and may get affected by other tests.

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala:1553]: It's a bit fragile to test `cacheManager` as it's global.

Can we create a new temp view with `path1` and check if it's cached?

**Comment by cloud-fan** [comment]
shall also we clean up the code in `CreateViewCommand`?

**Comment by sunchao** [comment]
Good catch on the cache issue! Unrelated: do we have documentation for the `ÇREATE OR REPLACE TEMP VIEW USING` clause? can't find it in [this page](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-view.html).

**Comment by imback82** [comment]
> shall also we clean up the code in `CreateViewCommand`?

Do you mean removing this condition? https://github.com/apache/spark/blob/9809a2f1c5187205c81542dbdc84b71db535f6e1/sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala#L628

If so, I am planning to address it with [SPARK-34700](https://issues.apache.org/jira/browse/SPARK-34700), in which we can change the signature of `getRawTempView`: https://github.com/imback82/spark-4/commit/f1c86b8e97b83d00a50d1e7421e93134c3afb5b8. I can still change it here in this PR, but SPARK-34700 will modify it a bit again. Please let me know which one you prefer. Thanks!

**Comment by cloud-fan** [comment]
@imback82 let's leave it to your follow-up work.

**Comment by imback82** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**